proposals/simd/SIMD.md

```diff
@@ -685,12 +698,32 @@ Load a single element and splat to all lanes of a `v128` vector.
 
 Fetch consequtive integers up to 32-bit wide and produce a vector with lanes up to 64 bits.
 
+```python
+def S.load_extend(ext, memarg):
+    result = S.New()
+    bytes = memory.bytes[memarg.start:memarg.start + 8])
+    for i in range(S.Lanes):
+        result[i] = ext(S.LaneType.from_byte(bytes[(i * S.LaneBytes/2):((i+1) * S.LaneBytes/2)]))
```

Comment:
Maybe something like `T.Load(memory, ...)` to make it clear that the semantics are the same as normal loads for value type `T`? I suppose `T` would have to be passed as an argument along with `ext`.

Reply:
The idea to have a single load, which gets broken up into lane values (though multiple loads should achieve the same binary result), that seems to be true-er to what actually would happen and also makes this snippet similar to snippets on other memory ops, which I feel is more instructive.